src/hotspot/cpu/aarch64/aarch64_sve.ad

```diff
@@ -340,7 +339,38 @@ instruct storeV_partial(vReg src, vmemA mem, pRegGov pTmp, iRegINoSp tmp1,
   %}
   ins_pipe(pipe_slow);
 %}
+instruct reinterpret(vReg dst) %{
+  predicate(UseSVE > 0 && n->as_Vector()->length_in_bytes() >= 16 &&
+            n->in(1)->bottom_type()->is_vect()->length_in_bytes() >= 16 &&
+            n->as_Vector()->length_in_bytes() ==
+            n->in(1)->bottom_type()->is_vect()->length_in_bytes());
+  match(Set dst (VectorReinterpret dst));
+  ins_cost(0);
+  format %{ " # reinterpret $dst" %}
+  ins_encode %{
+    // empty
+  %}
+  ins_pipe(pipe_class_empty);
+%}
 
+instruct reinterpretResize(vReg dst, vReg src) %{
+  predicate(UseSVE > 0 && n->as_Vector()->length_in_bytes() >= 16 &&
+            n->in(1)->bottom_type()->is_vect()->length_in_bytes() >= 16 &&
+            n->as_Vector()->length_in_bytes() !=
+            n->in(1)->bottom_type()->is_vect()->length_in_bytes());
+  match(Set dst (VectorReinterpret src));
+  ins_cost(SVE_COST);
+  format %{ " # reinterpret $dst,$src" %}
+  ins_encode %{
+    // If register is the same, then move is not needed.
+    if (as_FloatRegister($dst$$reg) != as_FloatRegister($src$$reg)) {
+      __ sve_orr(as_FloatRegister($dst$$reg),
+             as_FloatRegister($src$$reg),
+             as_FloatRegister($src$$reg));
+    }
+  %}
```

Comment:
Did you consider the resize case when the dst vector length in bytes is higher than the src vector length in bytes? I understand that for the parts that the length is `LE` to the src vector length, the values should be moved from the src values, otherwise the dst values are expected 0. Do you think it needs to use the SVE predicate feature to control the action here? 

For an example:
```
   1. byte[] a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]
   2. byte[] b = new byte[8];
       b = reinterpret(a) ;    // b = [1, 2, 3, 4, 5, 6, 7, 8] 
   3. byte[] c = new byte[16];
       c = reinterpret(b);     // c = [1, 2, 3, 4, 5, 6, 7, 8, 0, 0, 0, 0, 0, 0, 0, 0]
```
If no vector length info is referenced here, I'm wondering whether the final result is:
```
    c = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]
```